docks/crane_meta.go

```diff
@@ -0,0 +1,70 @@
+package docks
+
+import (
+	"time"
+
+	"github.com/safing/portbase/log"
+)
+
+const (
+	LaneLatencyTTL  = 1 * time.Hour
+	LaneCapacityTTL = 1 * time.Hour
+)
+
+// SetLaneLatency sets the lane latency to the given value.
+func (crane *Crane) SetLaneLatency(latency time.Duration) {
+	crane.metaLock.Lock()
+	defer crane.metaLock.Unlock()
+
+	crane.laneLatency = latency
+	crane.laneLatencyExpires = time.Now().Add(LaneLatencyTTL)
+
+	log.Infof("docks: %s lane latency set to %s", crane, latency)
+}
+
+// GetLaneLatency returns the lane latency.
+func (crane *Crane) GetLaneLatency() time.Duration {
+	crane.metaLock.Lock()
+	defer crane.metaLock.Unlock()
+
+	return crane.laneLatency
+}
+
+// LaneLatencyExpiresAt returns when the lane latency expires and should be
+// updated.
+func (crane *Crane) LaneLatencyExpiresAt() time.Time {
+	crane.metaLock.Lock()
+	defer crane.metaLock.Unlock()
+
+	return crane.laneLatencyExpires
+}
+
```

Comment:
I'd rather use one function instead of two and return both values. This way the caller must think about how he would handle outdated/stale data but still just do a `_` throw-away (`lat, _ := crane.LaneLatency()`)